ModelCatalogueGenomicsPlugin/src/groovy/org/modelcatalogue/gel/export/RareDiseaseChangeLogXlsExporter.groovy

```diff
@@ -338,101 +334,152 @@ abstract class RareDiseaseChangeLogXlsExporter extends AbstractChangeLogGenerato
         }
         deepElementValue
     }
 
-    //probably needs more protection
+
     private String extractChangedElement(String elemName) {
-        if (elemName && elemName.length() > 15) {   //some length indicating it might be a json string
+        if (mightBeJSON(elemName)) {
             def jsonMap
             def jsonSlurper = new JsonSlurper()
             try {
                 jsonMap = jsonSlurper.parseText(elemName)
             } catch (Exception e) {
                 log.error "Error attempting json parse of $elemName"
+                log.error ExceptionUtils.getStackTrace(e)
                 return elemName
             }
-            if (jsonMap && jsonMap.destination) {
-
-                String destination = jsonMap.destination.replaceAll(/[{}]/, '')
-                Map<String, String> destinationMap = new HashMap<String, String>().withDefault { ['', ''] }
-                def csvTokens = destination.split(',')
-                csvTokens.each { csvToken ->
-                    String[] tokens = csvToken.split('=')
-                    if (tokens.length < 2) {
-                        destinationMap.put(tokens[0], tokens[0])
+
+
+            try {
+                if (jsonMap && jsonMap.destination) {
+
+                    String destination = jsonMap.destination.replaceAll(/[{}]/, '')
+                    Map<String, String> destinationMap = new HashMap<String, String>().withDefault { ['', ''] }
+                    def csvTokens = destination.split(',')
+                    csvTokens.each { csvToken ->
+                        String[] tokens = csvToken.split('=')
+                        if (tokens.length < 2) {
+                            destinationMap.put(tokens[0], tokens[0])
+                        } else {
+                            destinationMap.put(tokens[0].trim(), tokens[1])
+                        }
+                    }
+                    if (destinationMap.classifiedName && destinationMap.classifiedName instanceof String) {
+                        elemName = "${destinationMap.classifiedName ?: ''} (${destinationMap.semanticVersion ?: ''})"
                     } else {
-                        destinationMap.put(tokens[0].trim(), tokens[1])
+                        elemName = ''
                     }
                 }
-                if(destinationMap.classifiedName && destinationMap.classifiedName instanceof String) {
-                    elemName = "${destinationMap.classifiedName ?: ''} (${destinationMap.semanticVersion ?: ''})"
-                } else {
-                    elemName = ''
-                }
+            } catch (Exception e) {
+                log.error "Error attempting csv token parse of $elemName"
+                log.error ExceptionUtils.getStackTrace(e)
+                return elemName
             }
         }
         elemName
     }
 
 
+    //is it potentially useful json? unlikely if less than 15 chars
+    public static boolean mightBeJSON(String jsonStr) {
```

Comment:
I'm aware of the readValue() method and had used it at one point but here I'm just trying to do a quick check to see if it's worth trying to parse it before I go ahead and try - inspired by the JsonUtils.maybeJson() method - rather than always relying on a thrown exception.

From: Vladimir Orany <notifications@github.com<mailto:notifications@github.com>>
Reply-To: MetadataRegistry/ModelCataloguePlugin <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Monday, 9 May 2016 at 13:27
To: MetadataRegistry/ModelCataloguePlugin <ModelCataloguePlugin@noreply.github.com<mailto:ModelCataloguePlugin@noreply.github.com>>
Cc: Rick Rees <rick.rees@genomicsengland.co.uk<mailto:rick.rees@genomicsengland.co.uk>>, Author <author@noreply.github.com<mailto:author@noreply.github.com>>
Subject: Re: [MetadataRegistry/ModelCataloguePlugin] ODA-833 improve parsing of metadata max/min occurs changes (#616)


In ModelCatalogueGenomicsPlugin/src/groovy/org/modelcatalogue/gel/export/RareDiseaseChangeLogXlsExporter.groovy<https://github.com/MetadataRegistry/ModelCataloguePlugin/pull/616#discussion_r62489570>:

>              }
>          }
>          elemName
>      }
>
>
> +    //is it potentially useful json? unlikely if less than 15 chars
> +    public static boolean mightBeJSON(String jsonStr) {


this might do the better job - https://github.com/MetadataRegistry/ModelCataloguePlugin/blob/e95528bf9e508f2326ed0304784e239ed3919158/ModelCatalogueCorePlugin/src/groovy/org/modelcatalogue/core/audit/LoggingAuditor.groovy#L426-L426

-
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/MetadataRegistry/ModelCataloguePlugin/pull/616/files/1fc6cfd77d4e090f084aa7ff167415ef70ef13e6#r62489570>